packages/atclient/include/atclient/connection.h

```diff
@@ -105,4 +131,31 @@ void atclient_connection_free(atclient_connection *ctx);
  */
 int atclient_connection_get_host_and_port(atclient_atstr *host, int *port, const atclient_atstr url);
 
+/**
+ * @brief Initialize the hooks memory allocation
+ *
+ * @param ctx the struct for the connection
+ */
+void atclient_connection_hooks_init(atclient_connection *ctx);
+
```

Comment:
We have `atclient_connection_hooks_init` function, but no `atclient_connection_hooks_free` function

Reply:
We don't need one, the hooks will be freed by atclient_connection_free